index.js

```diff
@@ -1,88 +1,119 @@
-// console.log("%cQuestion 1", "color: red") 
-
-// console.log(counter) 
+console.log("%cQuestion 1", "color: red")
+let counter = 1
+counter = 2
+console.log(counter)
 // // => 2
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 // ***** Question 2 *****
 // *** Uncomment the lines below to test
-// *** after testing, comment the line causing the error back in 
+// *** after testing, comment the line causing the error back in
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+console.log("%cQuestion 2", "color: red")
+const name = "James"
 
-// console.log(name)
+console.log(name)
 // => "Raffy"
-// name = "Not Raffy" 
+// name = "Not Raffy"
 // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 3 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red")
+
+function drinkWater(currentThirstLevel) {
+  console.log ("Man I sure am thirsty")
+  currentThirstLevel = currentThirstLevel - 1
+  console.log("Ahh that hits the spot")
+  return (currentThirstLevel)
+}
+
 
-// console.log(drinkWater(12))
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 4 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red")
 
-// console.log(sameSameButDifferent(5, 5)) 
+function sameSameButDifferent(num, maybeNum) {
+  if (num == parseInt(maybeNum)) {
+    if (typeof(num) == typeof(maybeNum)) {
+      return ("same same")
+    } else {
+      return ("same same (but different)")
+    }
+  }
+  if (num != parseInt(maybeNum)) {
+      return ("different")
+  }
+}
+console.log(sameSameButDifferent(5, 5))
 // // => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
+console.log(sameSameButDifferent(123, "123"))
 // // => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
+console.log(sameSameButDifferent(5, 7))
 // // => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
+console.log(sameSameButDifferent(123, "122"))
 // // => "different"
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 5 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade) {
+  student.grade = grade
+}
+
+
+
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
```

Comment:
Good 👍 The updates made to the `student` object inside the function is reflected in the `student1` object because objects are ["call by sharing" unlike primitive data types](https://stackoverflow.com/questions/518000/is-javascript-a-pass-by-reference-or-pass-by-value-language).